ios/FluentUI/Navigation/SideBar.swift

```diff
@@ -0,0 +1,284 @@
+		if numberOfItems > Constants.maxTabCount {
+			preconditionFailure("tab bar items can't be more than \(Constants.maxTabCount)")
+		}
+
+		let stackView = stackview(in: section)
+
+		for item in allItems {
+			let tabBarItemView = TabBarItemView(item: item, showsTitle: false)
+			let tapGesture = UITapGestureRecognizer(target: self, action: (section == .top) ? #selector(handleTopItemTapped(_:)) : #selector(handleBottomItemTapped(_:)))
+			tabBarItemView.addGestureRecognizer(tapGesture)
+
+			stackView.addArrangedSubview(tabBarItemView)
+		}
+
+		if (section == .top) {
+			selectedTopItem = allItems.first
+		}
+
+		setupLayout()
+	}
+
+	@objc open var selectedTopItem: TabBarItem? {
+		willSet {
+			if let item = selectedTopItem {
+				itemView(with: item, in: .top)?.isSelected = false
+			}
+		}
+		didSet {
+			if let item = selectedTopItem {
+				itemView(with: item, in: .top)?.isSelected = true
+			}
+		}
+	}
+
+	@objc public override init(frame: CGRect) {
+		super.init(frame: frame)
+		initialize()
+	}
+
+	@objc public required init?(coder: NSCoder) {
+		super.init(coder: coder)
+		initialize()
+	}
+
+	/// Helper method to insert the view in a superview.
+	@objc public func insert(in superView: UIView) {
+		superView.addSubview(self)
+
+		NSLayoutConstraint.activate([
+			leadingAnchor.constraint(equalTo: superView.leadingAnchor),
+			topAnchor.constraint(equalTo: superView.topAnchor),
+			bottomAnchor.constraint(equalTo: superView.bottomAnchor)
+		])
+	}
+
+	private let backgroundView: UIVisualEffectView = {
+		var style = UIBlurEffect.Style.regular
+		if #available(iOS 13, *) {
+			style = .systemMaterial
+		}
+
+		return UIVisualEffectView(effect: UIBlurEffect(style: style))
+	}()
+
+	private let borderLine = Separator(style: .shadow, orientation: .vertical)
+
+	private func initialize() {
+		translatesAutoresizingMaskIntoConstraints = false
+
+		backgroundView.translatesAutoresizingMaskIntoConstraints = false
+		contain(view: backgroundView)
+
+		borderLine.translatesAutoresizingMaskIntoConstraints = false
+		addSubview(borderLine)
+
+		addSubview(topStackView)
+		addSubview(bottomStackView)
+
+		if #available(iOS 13, *) {
+			addInteraction(UILargeContentViewerInteraction())
+		}
+
+		accessibilityTraits = .tabBar
+
+		NSLayoutConstraint.activate([widthAnchor.constraint(equalToConstant: Constants.viewWidth),
+									 borderLine.leadingAnchor.constraint(equalTo: trailingAnchor),
+									 borderLine.bottomAnchor.constraint(equalTo: bottomAnchor),
+									 borderLine.topAnchor.constraint(equalTo: topAnchor)])
+	}
+
+	private var layoutConstraints: [NSLayoutConstraint] = []
+
+	private func setupLayout() {
+		if layoutConstraints.count > 0 {
+			NSLayoutConstraint.deactivate(layoutConstraints)
+			layoutConstraints.removeAll()
+		}
+
+		if let avatarView = self.avatarView {
+			layoutConstraints.append(contentsOf: [
+				avatarView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 18.0),
```

Comment:
pull the constants (18.0 here, others below) out of line into constants to make them self-documenting